applications/vanilla/models/class.discussionmodel.php

```diff
@@ -2464,4 +2464,38 @@ public static function allowedSortFields($Allowed = null) {
 
         return self::$AllowedSortFields;
     }
+
+
+    /**
+     * Tests whether a user has permission to view a specific discussion by checking category-specific permissions.
+     * Fires an event that can override the category view permission.
+     *
+     * @param object|array|integer $discussion The discussion ID or the discussion to test.
+     * @param string|integer $userID The ID of the user to test permission for. If empty, it defaults to Session user.
+     * @param string $categoryPermission The category permission to test against the user.
+     * @return bool Whether the user can view the discussion.
+     * @throws Exception
+     */
+    public function canView($discussion, $userID = '', $categoryPermission = 'Vanilla.Discussions.View') {
```

Comment:
Using `$userID = 0` will allow this method to have the userID be just an integer which is a move we want to make to help hint to future compilers.